<a href="https://colab.research.google.com/github/Ruthnguli/Skilled-Based-Test-IMdb/blob/main/Rut_Nguli_imdb_skilled_based_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
# mounting drive to access the imdb data set
#
from google.colab import drive 
drive.mount('/content/drive')


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [6]:
# importing libraries to load the imdb zip file from google drive
#
import pandas as pd
import zipfile
with zipfile.ZipFile('/content/drive/MyDrive/imdb/archive.zip', 'r') as zip_ref:
  zip_ref.extractall('/content/drive/MyDrive/imdb')

In [7]:
# reading the json files
#
data=pd.read_json("/content/drive/MyDrive/imdb/part-01.json")

In [8]:
# creating a data frame
#
reviews_df = pd.DataFrame(data)

In [9]:
# cheching the data shape
#
reviews_df.shape

(1010293, 9)

In [10]:
#preview of the top 5
#
reviews_df.head()

review_id         reviewer                          movie  rating  \
0  rw5704482    raeldor-96879            After Life (2019– )     9.0   
1  rw5704483          dosleeb  The Valhalla Murders (2019– )     6.0   
2  rw5704484  brightconscious           Special OPS (2020– )     7.0   
3  rw5704485       gasconyway              #BlackAF (2020– )     8.0   
4  rw5704487     mmason-15867             The Droving (2020)     2.0   

          review_summary review_date  spoiler_tag  \
0   Very Strong Season 2  3 May 2020            0   
1  Icelandic detectives?  3 May 2020            0   
2        Nothing special  3 May 2020            0   
3               Good but  3 May 2020            0   
4       An honest review  3 May 2020            0   

                                       review_detail   helpful  
0  I enjoyed the first season, but I must say I t...    [1, 1]  
1  I know Iceland is a small country and police d...    [2, 2]  
2  Except K K , no other actor looks comfortable ...    [0, 0]  
3  I'm guessing that as a 62 year old white woman...    [5, 9]  
4  Here's the truth. There's not much to this mov...  [26, 41]

In [11]:
# Checking Null values
#
reviews_df.isna().sum()

review_id             0
reviewer              0
movie                 0
rating            51520
review_summary        0
review_date           0
spoiler_tag           0
review_detail         0
helpful               0
dtype: int64

In [12]:
# dropping missing values and confirming if missing has been dropped
#
df = reviews_df.dropna()

df.isna().sum()

review_id         0
reviewer          0
movie             0
rating            0
review_summary    0
review_date       0
spoiler_tag       0
review_detail     0
helpful           0
dtype: int64

## Questions

1. Which movie had the highest rating?
2. How many reviewers gave it the highest rating?
3. What’s its least rating and review detail?

In [13]:
# A pie chart of the ratings 
#
ratings = reviews_df["rating"].value_counts()
numbers = ratings.index
quantity = ratings.values
import plotly.express as px
fig = px.pie(reviews_df, values=quantity, names=numbers)
fig.show()

In [14]:
# grouping the movie and ratings to get the total ratings per movie
#
ratings = reviews_df.groupby(['movie', 'rating']).size().reset_index(name='total_ratings')
ratings.sort_values(by=['rating','total_ratings'],ascending=[False,False],inplace=True)

ratings.head(5)

movie  rating  total_ratings
68130                          Dil Bechara (2020)    10.0           7087
309626                                  小丑 (2019)    10.0           4062
197579  Scam 1992: The Harshad Mehta Story (2020)    10.0           2764
158516                  Mrs. Serial Killer (2020)    10.0           2312
209528                     Soorarai Pottru (2020)    10.0           1299

Dil Bechara (2020) is the highest rated movie with 7087 giving it high ratings

In [17]:
# finding the unique reveiw to th highest rating
#
unique_rating = ratings.loc[ratings['movie'] == 'Dil Bechara (2020)', 'rating'].max() 

reviewers = reviews_df[(reviews_df['movie'] == 'Dil Bechara (2020)') & (reviews_df['rating'] == unique_rating)].reviewer.nunique()
reviewers
#print(f"{reviewers_count} uniquie reviewers gave the highest rating to the top movie")

7033

7033 reviewers gave it the highest rating

In [22]:
# Finding the least rating and its review
#

least_df = reviews_df.loc[reviews_df['movie'] == 'Dil Bechara (2020)', 'rating'].min()
lst = reviews_df[(reviews_df['movie'] == 'Dil Bechara (2020)') & (reviews_df['rating'] == least_df)]
least_rated = lst.tail(1)
rating = least_rated['rating'].values[0]
review_detail = least_rated['review_detail'].values[0]
print(rating)

print(                       )

print(review_detail)

1.0

Review Detail:  Imagine if a drug addict and womanizing actor had not died a dog's death would this movie get a 7.9 rating. I remember when this chap got a 10 rating when it was realized but time is a healer..now that that Bihar elections are over and its nearly proved that the drug addict SSR was a loser ,this movie has no solid to hold.
For starters it's a rip off with over acting and minority mocking .
The acting of the womanizing dead druggie actor is noting but mediocre.


Least rating is 1. 

Review detail is " Imagine if a drug addict and womanizing actor had not died a dog's death would this movie get a 7.9 rating. I remember when this chap got a 10 rating when it was realized but time is a healer..now that that Bihar elections are over and its nearly proved that the drug addict SSR was a loser ,this movie has no solid to hold.
For starters it's a rip off with over acting and minority mocking .
The acting of the womanizing dead druggie actor is noting but mediocre."